### DISCUSSION TITLE
Diablo 2 Resurrected settings and local data lost after Flatpak Bootles restart

### DISCUSSION THREAD

**Discussion by stonebuckl** [bottlesdevs/Bottles#571] [Q&A] [answered]
I have an issue where after closing and opening Bottles running with Flatpak all settings and local characters in Diablo 2 Resurrected are deleted. It seems as if those files are stored in a virtual file system that only exists during the sandboxed execution. I also understand that I could probably expose a folder to circumvent this issue as described [here](https://docs.usebottles.com/flatpak/expose-directories). However, I think it should be possible for D2R to create and keep local files _inside_ the sandbox. Am I wrong?

It seems to me, the main issue is that in my Bottles `winecfg`, the folder `Documents` links to `/home/username` instead of (what I would assume to be correct) an existing directory in the bottle directory. Using `strace`, I can see that the files are accessed at `/home/username/.var/app/com.usebottles.bottles/data/bottles/bottles/Battlenet/dosdevices/c:/users/username/Documents/Diablo II Resurrected/Save/` but that folder only exists in Wine (checked with command prompt), not physically on the disk. Consequently, the folder vanishes after closing Bottles.

```
[pid 80501] 20:56:17 lstat("/home/username/.var/app/com.usebottles.bottles/data/bottles/bottles/Battlenet/dosdevices/c:/users/username/Documents/Diablo II Resurrected/Save/Isabella.d2s", {st_mode=S_IFREG|0644, st_size=945, ...}) = 0
```

My understanding is that as the Battle.net client can somehow use local files to keep my login credentials and store game data, so should the game itself be able to. Am I wrong?

**Comment by mirkobrombin** [comment, marked as answer]
I've managed to reproduce the problem, using notepad I created a file and saved in `C:\users\USER\Downloads`, the file is detected by wine but not by the system and once the subsystem is restarted, the file disappears.

The problem is that wine default point the folders inside the user's dir, to the user's home dir. As you have seen, you can fix the problem by exposing the homedir to Bottles, so that the runner can write to that directory.

Regarding your question, I believe it is possible to force wine to use an internal path for those links, I believe I will turn this thread into an issue to add the functionality to the roadmap.


https://github.com/bottlesdevs/Bottles/issues/572

**Reply by stonebuckl** [reply]
Much appreciated!
